provd/persist/util.py

```diff
@@ -552,3 +551,22 @@ def __init__(self, collection):
 
     def __getattr__(self, name):
         return getattr(self._collection, name)
+
+
+def _new_key_fun_from_key(key):
+    """
+    Return a function usable for the key parameter of the sorted function
+    from a [sort] key
+    """
+    split_key = key.split('.')
+
+    def func(document):
+        cur_elem = document
+        try:
+            for cur_key in split_key:
+                cur_elem = cur_elem[cur_key]
+        except (KeyError, TypeError) as e:
+            return ''
+        return str(cur_elem)
```

Comment:
Did you check what `cur_elem` was when it failed? Can it ever be None? Integers are fine to convert to string, but instead of `'None'` we would probably prefer `''` in that case too.